src/state_machine.zig

```diff
@@ -1006,8 +1006,8 @@ pub fn StateMachineType(
                 cr_account_new.credits_posted += amount;
             }
 
-            self.forest.grooves.accounts.upsert(&dr_account_new);
-            self.forest.grooves.accounts.upsert(&cr_account_new);
+            self.forest.grooves.accounts.update(&dr_account_new, dr_account);
```

Comment:
A risk to my suggestion here though is that if an `insert(new)` is swapped to an `update(old, new)` then the caller needs to swap arguments.

However, if the interface remains `update(new, old)` then there is the risk that the caller does `(old, new)`... thinking progress moves positively, from left to right.

Reply:
Yeah indeed - my ordering was based off the first case case, which also imitates our `target` first style for copies... A third option could be to make it an explicit struct? 

```zig
self.forest.grooves.accounts.update(.{ .old = dr_account, .new = &dr_account_new });
```

From what I can see, we don't have any other `update` functions floating around to base the style off.

> Two parameters of the same type here makes me nervous 😅

~ DJ, 1 year ago, on my AOF PR :smile:

Reply:
Yes, for copies, we should also rather be moving from left to right, from `source` to `target` (I know the Zig std lib does it the other way, but we can break rhyme).

Reply:
After a bit more discussion:
- `target` first style for copies gets `const` protection, which we can't take advantage of here.
- There's no real convention, because this `.update()` method doesn't update a parameter with an existing one, but rather updates the groove based off of both.
- There's no performance cost to using a struct (asm generated on godbolt was identical and double checked on the benchmark).

I've gone with a struct.